pkg/build/strategies/sti/sti.go

```diff
@@ -215,13 +215,18 @@ func (builder *STI) Build(config *api.Config) (*api.Result, error) {
 		}
 	}
 
-	if len(config.AssembleUser) > 0 {
-		glog.V(1).Infof("Running %q in %q as %q user", api.Assemble, config.Tag, config.AssembleUser)
+	assembleUser, err := builder.docker.GetAssembleUser(config.AssembleUser, config.BuilderImage)
+	if err != nil {
+		return builder.result, err
+	}
+
+	if len(assembleUser) > 0 {
+		glog.V(1).Infof("Running %q in %q as %q user", api.Assemble, config.Tag, assembleUser)
 	} else {
 		glog.V(1).Infof("Running %q in %q", api.Assemble, config.Tag)
 	}
 	startTime := time.Now()
-	if err := builder.scripts.Execute(api.Assemble, config.AssembleUser, config); err != nil {
+	if err := builder.scripts.Execute(api.Assemble, assembleUser, config); err != nil {
```

Comment:
https://github.com/openshift/source-to-image/blob/master/pkg/docker/util.go#L304

which calls https://github.com/openshift/source-to-image/blob/master/pkg/docker/util.go#L231-L240